src/x86.rs

```diff
@@ -434,57 +443,82 @@ impl DynamicAvx2Searcher {
     /// for `needle`.
     #[target_feature(enable = "avx2")]
     pub unsafe fn with_position(needle: Box<[u8]>, position: usize) -> Self {
-        match *needle {
-            [] => Self::N0,
-            [c0] => {
+        #[inline]
+        fn slice_to_array<const N: usize>(src: &[u8]) -> [u8; N] {
+            use std::mem::MaybeUninit;
+
+            let mut arr = MaybeUninit::<[u8; N]>::uninit();
+            unsafe { &mut *arr.as_mut_ptr() }.copy_from_slice(src);
+            unsafe { arr.assume_init() }
+        }
```

Comment:
I'm not sure about this but could we avoid the unsafe code by just using `.try_into().unwrap()` as arrays implement `TryFrom<&[T]>`?

Reply:
Technically, that's what we had before, but I would want to verify codegen because for the oneshot usecases, I would this to be fast so I would want the actually unwrap etc to be removed by the compiler

Reply:
Ah okay, I see. It might be an option to use [`unreachable_unchecked()`](https://doc.rust-lang.org/std/hint/fn.unreachable_unchecked.html) (although that is unsafe in itself) but I'll leave it to you to decide what's best :)

Reply:
I've proposed another approach in #45

Reply:
Cool, I think that approach is great!